crates/iota/src/upgrade_compatibility.rs

```diff
@@ -1362,10 +1363,64 @@ fn struct_ability_mismatch_diag(
     Ok(diags)
 }
 
+/// Return a diagnostic for an ability mismatch. returns (full version, name,
+/// type)
+fn field_to_string(field: &Field) -> (String, String, String) {
+    let mut field_full = format!("'{}: {}'", field.name, field.type_);
+    let mut field_name = format!("'{}'", field.name);
+    let field_type = format!("'{}'", field.type_);
+
+    if let Some(pos_num) = Regex::new(r"^pos(\d)+$")
+        .ok()
+        .and_then(|r| r.captures(field.name.as_str()))
+        .and_then(|c| c.get(1))
+        .and_then(|m| m.as_str().parse::<u64>().ok())
+    {
+        field_name = format!("at position {}", pos_num);
+        field_full = format!("{} {}", field_type, field_name);
+    }
+
+    (field_full, field_name, field_type)
+}
+
+/// returns a message for the given field
+fn field_mismatch_message(old_field: &Field, new_field: &Field) -> (Declarations, String) {
+    let (old_field_full, old_field_name, old_field_type) = field_to_string(old_field);
+    let (new_field_full, new_field_name, new_field_type) = field_to_string(new_field);
+
+    match (
+        old_field.name != new_field.name,
+        old_field.type_ != new_field.type_,
+    ) {
+        (true, true) => (
+            Declarations::FieldMismatch,
+            format!(
+                "Mismatched field {}, expected {}.",
+                new_field_full, old_field_full
+            ),
+        ),
+        (true, false) => (
+            Declarations::FieldMismatch,
+            format!(
+                "Mismatched field name {}, expected {}.",
+                new_field_name, old_field_name
+            ),
+        ),
+        (false, true) => (
+            Declarations::TypeMismatch,
+            format!(
+                "Mismatched field type {}, expected {}.",
+                new_field_type, old_field_type
+            ),
+        ),
+        (false, false) => unreachable!("Fields should not be the same"),
+    }
+}
+
 /// Return a diagnostic for a field mismatch
-/// start by checking the lengths of the fields and return a diagnostic if they
-/// are different if the lengths are the same check each field piece wise and
-/// return a diagnostic for each mismatch
+/// Start by checking the lengths of the fields and return a diagnostic if they
+/// are different. If the lengths are the same check each field piece wise and
+/// return a diagnostic for each mismatch.
```

Comment:
That's the type of comment we changed in a previous PR because it's not very well worded. Not sure if it was undone accidentally or if it's a doc on a different function.

Reply:
`function_signature_mismatch_diag` here it was

Reply:
Propose something if you want but I'd rather do the least amount of changes on the pullings and do changes on the final PR instead

Reply:
fine with me ... let's do it in the final PR then. We can use the docs from that function I mentioned above here. maybe there are other cases as well, where this comment wording was used, but we can see in the final PR.